# **Assignment** # 1

 **Name:** Lama Batta 

**ID Student:** 1210922

<span style="color:blue"> **Import Libraries** </span>

In [25]:
import pandas as pd

**Load the data set**

In [26]:
# Read data from csv file
df = pd.read_csv('assignment_1-python.csv')
df

date      region        city      market  latitude  longitude  \
0    2022-01-15  Gaza Strip  Kan Younis  Gaza Strip  31.34611   34.32573   
1    2022-01-15  Gaza Strip  Kan Younis  Gaza Strip  31.34611   34.32573   
2    2022-01-15  Gaza Strip  Kan Younis  Gaza Strip  31.34611   34.32573   
3    2022-01-15  Gaza Strip  Kan Younis  Gaza Strip  31.34611   34.32573   
4    2022-01-15  Gaza Strip  Kan Younis  Gaza Strip  31.34611   34.32573   
..          ...         ...         ...         ...       ...        ...   
349  2022-12-15  Gaza Strip  Kan Younis  Gaza Strip  31.34611   34.32573   
350  2022-12-15  Gaza Strip  Kan Younis  Gaza Strip  31.34611   34.32573   
351  2022-12-15  Gaza Strip  Kan Younis  Gaza Strip  31.34611   34.32573   
352  2022-12-15  Gaza Strip  Kan Younis  Gaza Strip  31.34611   34.32573   
353  2022-12-15  Gaza Strip  Kan Younis  Gaza Strip  31.34611   34.32573   

                  category                        commodity   unit priceflag  \
0       cereals and tubers                            Bread     KG    actual   
1       cereals and tubers           Potatoes (medium size)     KG    actual   
2       cereals and tubers     Rice (small grain, imported)  25 KG    actual   
3       cereals and tubers  Wheat flour (locally processed)  50 KG    actual   
4      meat, fish and eggs                             Eggs   2 KG    actual   
..                     ...                              ...    ...       ...   
349  vegetables and fruits                      Cauliflower     KG    actual   
350  vegetables and fruits           Cucumbers (greenhouse)     KG    actual   
351  vegetables and fruits                Eggplants (large)     KG    actual   
352  vegetables and fruits              Onions (dry, local)     KG    actual   
353  vegetables and fruits            Tomatoes (greenhouse)     KG    actual   

    pricetype currency   price  usdprice  
0      Retail      ILS    2.50    0.7934  
1      Retail      ILS    1.94    0.6157  
2      Retail      ILS  152.31   48.3351  
3      Retail      ILS   90.00   28.5612  
4      Retail      ILS   11.72    3.7193  
..        ...      ...     ...       ...  
349    Retail      ILS    2.95    0.8626  
350    Retail      ILS    1.83    0.5351  
351    Retail      ILS    1.95    0.5702  
352    Retail      ILS    2.06    0.6023  
353    Retail      ILS    2.92    0.8538  

[354 rows x 14 columns]

<span style="color:red"> **Identify the most expensive vegetables and fruits commodity during this period.** </span>

In [27]:
# Filter for vegetables and fruits
vegetables_and_fruits=df[df['category']=='vegetables and fruits']
vegetables_and_fruits

date      region        city      market  latitude  longitude  \
23   2022-01-15  Gaza Strip  Kan Younis  Gaza Strip  31.34611   34.32573   
24   2022-01-15  Gaza Strip  Kan Younis  Gaza Strip  31.34611   34.32573   
25   2022-01-15  Gaza Strip  Kan Younis  Gaza Strip  31.34611   34.32573   
26   2022-01-15  Gaza Strip  Kan Younis  Gaza Strip  31.34611   34.32573   
27   2022-01-15  Gaza Strip  Kan Younis  Gaza Strip  31.34611   34.32573   
..          ...         ...         ...         ...       ...        ...   
349  2022-12-15  Gaza Strip  Kan Younis  Gaza Strip  31.34611   34.32573   
350  2022-12-15  Gaza Strip  Kan Younis  Gaza Strip  31.34611   34.32573   
351  2022-12-15  Gaza Strip  Kan Younis  Gaza Strip  31.34611   34.32573   
352  2022-12-15  Gaza Strip  Kan Younis  Gaza Strip  31.34611   34.32573   
353  2022-12-15  Gaza Strip  Kan Younis  Gaza Strip  31.34611   34.32573   

                  category               commodity unit priceflag pricetype  \
23   vegetables and fruits            Apples (red)   KG    actual    Retail   
24   vegetables and fruits   Bananas (medium size)   KG    actual    Retail   
25   vegetables and fruits             Cauliflower   KG    actual    Retail   
26   vegetables and fruits  Cucumbers (greenhouse)   KG    actual    Retail   
27   vegetables and fruits     Onions (dry, local)   KG    actual    Retail   
..                     ...                     ...  ...       ...       ...   
349  vegetables and fruits             Cauliflower   KG    actual    Retail   
350  vegetables and fruits  Cucumbers (greenhouse)   KG    actual    Retail   
351  vegetables and fruits       Eggplants (large)   KG    actual    Retail   
352  vegetables and fruits     Onions (dry, local)   KG    actual    Retail   
353  vegetables and fruits   Tomatoes (greenhouse)   KG    actual    Retail   

    currency  price  usdprice  
23       ILS   4.59    1.4566  
24       ILS   3.02    0.9584  
25       ILS   2.30    0.7299  
26       ILS   3.58    1.1361  
27       ILS   2.23    0.7077  
..       ...    ...       ...  
349      ILS   2.95    0.8626  
350      ILS   1.83    0.5351  
351      ILS   1.95    0.5702  
352      ILS   2.06    0.6023  
353      ILS   2.92    0.8538  

[75 rows x 14 columns]

<span style="color:blue">This is the result of filtering a dataset for the category Fruits and Vegetables, where there are 75 rows containing this category.  </span>

In [38]:
#find the most expensive price vegetables and fruits 

most_expensive=vegetables_and_fruits[vegetables_and_fruits['usdprice']==vegetables_and_fruits['usdprice'].max()].iloc[0]
print(f"Commodity: {most_expensive['commodity']}, Price: {most_expensive['usdprice']}")


Commodity: Apples (red), Price: 1.7703


<span style="color:blue">This is the result of the most expensive commodity of vegetables and fruits, which is apples, and its value price is 1.7703
  </span>

<span style="color:red"> **Add a new column named currency_rate that captures the exchange rate between the local currency and the US dollar (USD).** </span>

In [29]:
#create new column currency_rate
df['currency_rate']=df['price']/df['usdprice']
df

date      region        city      market  latitude  longitude  \
0    2022-01-15  Gaza Strip  Kan Younis  Gaza Strip  31.34611   34.32573   
1    2022-01-15  Gaza Strip  Kan Younis  Gaza Strip  31.34611   34.32573   
2    2022-01-15  Gaza Strip  Kan Younis  Gaza Strip  31.34611   34.32573   
3    2022-01-15  Gaza Strip  Kan Younis  Gaza Strip  31.34611   34.32573   
4    2022-01-15  Gaza Strip  Kan Younis  Gaza Strip  31.34611   34.32573   
..          ...         ...         ...         ...       ...        ...   
349  2022-12-15  Gaza Strip  Kan Younis  Gaza Strip  31.34611   34.32573   
350  2022-12-15  Gaza Strip  Kan Younis  Gaza Strip  31.34611   34.32573   
351  2022-12-15  Gaza Strip  Kan Younis  Gaza Strip  31.34611   34.32573   
352  2022-12-15  Gaza Strip  Kan Younis  Gaza Strip  31.34611   34.32573   
353  2022-12-15  Gaza Strip  Kan Younis  Gaza Strip  31.34611   34.32573   

                  category                        commodity   unit priceflag  \
0       cereals and tubers                            Bread     KG    actual   
1       cereals and tubers           Potatoes (medium size)     KG    actual   
2       cereals and tubers     Rice (small grain, imported)  25 KG    actual   
3       cereals and tubers  Wheat flour (locally processed)  50 KG    actual   
4      meat, fish and eggs                             Eggs   2 KG    actual   
..                     ...                              ...    ...       ...   
349  vegetables and fruits                      Cauliflower     KG    actual   
350  vegetables and fruits           Cucumbers (greenhouse)     KG    actual   
351  vegetables and fruits                Eggplants (large)     KG    actual   
352  vegetables and fruits              Onions (dry, local)     KG    actual   
353  vegetables and fruits            Tomatoes (greenhouse)     KG    actual   

    pricetype currency   price  usdprice  currency_rate  
0      Retail      ILS    2.50    0.7934       3.150996  
1      Retail      ILS    1.94    0.6157       3.150885  
2      Retail      ILS  152.31   48.3351       3.151126  
3      Retail      ILS   90.00   28.5612       3.151128  
4      Retail      ILS   11.72    3.7193       3.151131  
..        ...      ...     ...       ...            ...  
349    Retail      ILS    2.95    0.8626       3.419893  
350    Retail      ILS    1.83    0.5351       3.419922  
351    Retail      ILS    1.95    0.5702       3.419853  
352    Retail      ILS    2.06    0.6023       3.420222  
353    Retail      ILS    2.92    0.8538       3.420005  

[354 rows x 15 columns]

<span style="color:blue">This is the dataset after we calculated the exchange rate between the local currency and the US dollar (USD) for each row and then created a new column named currency_rate containing these values.
  </span>

<span style="color:red"> **Create a new column called unified_price that represents the price in USD per kilogram for commodities measured by weight and the price per liter for commodities measured by volume.** </span>

In [134]:
# Custom function to calculate  unified_price
def unified_price(row):
    #split unit column for each row 
    Value_unit= row['unit'].split()
    value = 1.0  
    if len(Value_unit) > 1 and Value_unit[0].isdigit():
        value = float(Value_unit[0]) 
    elif len(Value_unit) == 1 and Value_unit[0].isdigit():
        value = float(Value_unit[0])
        
    if 'KG' in row['unit']:
        return row['usdprice'] /value
    elif 'G' in row['unit']:
        return row['usdprice']  /(value/ 1000 )
    elif 'L' in row['unit']:
        return row['usdprice'] /value 
    elif 'Cubic meter' in row['unit']:
        return row['usdprice'] /(value * 1000 )
    else:
        return None
# Applying the function to each row
df['unified_price'] = df.apply(unified_price, axis=1)
df

date      region        city      market  latitude  longitude  \
0    2022-01-15  Gaza Strip  Kan Younis  Gaza Strip  31.34611   34.32573   
1    2022-01-15  Gaza Strip  Kan Younis  Gaza Strip  31.34611   34.32573   
2    2022-01-15  Gaza Strip  Kan Younis  Gaza Strip  31.34611   34.32573   
3    2022-01-15  Gaza Strip  Kan Younis  Gaza Strip  31.34611   34.32573   
4    2022-01-15  Gaza Strip  Kan Younis  Gaza Strip  31.34611   34.32573   
..          ...         ...         ...         ...       ...        ...   
349  2022-12-15  Gaza Strip  Kan Younis  Gaza Strip  31.34611   34.32573   
350  2022-12-15  Gaza Strip  Kan Younis  Gaza Strip  31.34611   34.32573   
351  2022-12-15  Gaza Strip  Kan Younis  Gaza Strip  31.34611   34.32573   
352  2022-12-15  Gaza Strip  Kan Younis  Gaza Strip  31.34611   34.32573   
353  2022-12-15  Gaza Strip  Kan Younis  Gaza Strip  31.34611   34.32573   

                  category                        commodity   unit priceflag  \
0       cereals and tubers                            Bread     KG    actual   
1       cereals and tubers           Potatoes (medium size)     KG    actual   
2       cereals and tubers     Rice (small grain, imported)  25 KG    actual   
3       cereals and tubers  Wheat flour (locally processed)  50 KG    actual   
4      meat, fish and eggs                             Eggs   2 KG    actual   
..                     ...                              ...    ...       ...   
349  vegetables and fruits                      Cauliflower     KG    actual   
350  vegetables and fruits           Cucumbers (greenhouse)     KG    actual   
351  vegetables and fruits                Eggplants (large)     KG    actual   
352  vegetables and fruits              Onions (dry, local)     KG    actual   
353  vegetables and fruits            Tomatoes (greenhouse)     KG    actual   

    pricetype currency   price  usdprice  currency_rate  unified_price  
0      Retail      ILS    2.50    0.7934       3.150996       0.793400  
1      Retail      ILS    1.94    0.6157       3.150885       0.615700  
2      Retail      ILS  152.31   48.3351       3.151126       1.933404  
3      Retail      ILS   90.00   28.5612       3.151128       0.571224  
4      Retail      ILS   11.72    3.7193       3.151131       1.859650  
..        ...      ...     ...       ...            ...            ...  
349    Retail      ILS    2.95    0.8626       3.419893       0.862600  
350    Retail      ILS    1.83    0.5351       3.419922       0.535100  
351    Retail      ILS    1.95    0.5702       3.419853       0.570200  
352    Retail      ILS    2.06    0.6023       3.420222       0.602300  
353    Retail      ILS    2.92    0.8538       3.420005       0.853800  

[354 rows x 16 columns]

<span style="color:blue">This is the dataset after we calculated the unified price, so we created a new column called unified_price and the results were as follows: price in USD per kilogram, price per liter for each commodity measured by weight or volume
  </span>

<span style="color:red"> **Using the unified_price, find the maximum price reached during this period and for which commodity for both:
commodities measured by weight
commodities measured by volume** </span>

In [131]:
#find the maximum price for each commodity measured by weight
max_unified_price = df[(df['unit'] == 'KG') | (df['unit'] == 'G')]['unified_price'].max() 
maximum_weight = df[df['unified_price'] == max_unified_price].iloc[0]
print(f"Commodity: {maximum_weight['commodity']}, Unified Price: {maximum_weight['unified_price']}")

Commodity: Meat (goat, with bones), Unified Price: 17.3144


<span style="color:blue">This is the result of the maximum price commodity measured by weight, which is Meat (goat, with bones),and its value Unified Price: 17.3144 which is the maximum price of commodities measured by weight.
  </span>

In [132]:
#find the maximum price for each commodity measured by volume
max_unified_price = df[(df['unit'] == 'L') | (df['unit'] == 'Cubic meter')]['unified_price'].max()
maximum_volume = df[df['unified_price'] == max_unified_price].iloc[0]
print(f"Commodity: {maximum_volume['commodity']}, Unified Price: {maximum_volume['unified_price']}")

Commodity: Fuel (petrol-gasoline), Unified Price: 2.0459


<span style="color:blue">This is the result of the maximum price commodity measured by volume, which is Fuel (petrol-gasoline),and its value Unified Price:  2.0459 which is the maximum price of commodities measured by volume.
  </span>